### ISSUE TITLE
twisted.conch.client.default should be ported to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8700]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)| @einarfd reported|
|-|-|
|Trac ID|trac#8700|
|Type|enhancement|
|Created|2016-07-28 08:52:43Z|
|Branch|https://github.com/twisted/twisted/tree/8700-einarfd-py3-conch-client-default-3|

<details><summary>Searchable metadata</summary>

```
trac-id__8700 8700
type__enhancement enhancement
reporter__einarfd einarfd
priority__normal normal
milestone__None None
branch__8700_einarfd_py3_conch_client_default_3 8700-einarfd-py3-conch-client-default-3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469695963226765 1469695963226765
changetime__1470602005356318 1470602005356318
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="einarfd's avatar" src="https://avatars.githubusercontent.com/u/5199247?s=50" width="50" height="50">](https://github.com/einarfd)<a name="note_1"></a>|@einarfd removed owner|
|-|-|

Added a merge request for this at https://github.com/twisted/twisted/pull/438 .

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_2"></a>|@rodrigc set owner to @rodrigc<br>@rodrigc set status to `closed`|
|-|-|

In changeset c87981104d0fa1bbfcabceae63256ea0d79c42c1
```
#!CommitTicketReference repository="" revision="c87981104d0fa1bbfcabceae63256ea0d79c42c1"
Merge branch '8700-einarfd-py3-conch-client-default' into trunk

Author: einarfd
Reviewer: rodrigc
Fixes: #8700

Port twisted.conch.client.default to Python 3
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

In changeset 4330d2c1191b5e8fa43579eb0accb6d093c47dbf
```
#!CommitTicketReference repository="" revision="4330d2c1191b5e8fa43579eb0accb6d093c47dbf"
Revert "Merge branch '8700-einarfd-py3-conch-client-default' into trunk"

Reopens: #8700

This reverts commit c87981104d0fa1bbfcabceae63256ea0d79c42c1, reversing
changes made to b636ffb4afba9f1c30b91ddef95e4cef38c2dd26.
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set status to `reopened`|
|-|-|

Hrm. I guess the "reopens" tag (described here: https://twistedmatrix.com/trac/wiki/ReviewProcess#Revertingachange) doesn't work any more?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

Hi Craig,

Thanks for reviewing this.  Looks like we have some very subtle failures.

In the future, please:

1. remove the 'review' keyword from the ticket before merging - mostly so you get the credit on the scoreboard :).
2. push the change to a named branch on github to ensure that the buildbots get a crack at it first.  The buildbot github status stuff will very helpfully put the commit statuses on the PR as long as you push the same commit to a named branch.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc removed owner<br>@rodrigc set status to `new`|
|-|-|

Glyph,

I have created a new pull request which brings back the changes from 8700-einarfd-py3-conch-client-default
and adds a small fix to the tests which succeed on the buildbot.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph set owner to @rodrigc|
|-|-|

Hey rodrigc; thanks for following up.  I did another code review and I found a few issues which I left as github comments, but they are mostly cosmetic.  Please address them to your satisfaction and land.  Another module down!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph commented|
|-|-|

(... "land assuming all the buildbots pass again" of course :))

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_9"></a>|@rodrigc set status to `closed`|
|-|-|

Fixed in 6c6fea1bc2107601a2d270f2ee709ca7e8a24468
